Complete hw2:

stevensonmat2 opened the pull request:
- implement js to highlight text for count exercise
- add bootstrap styling to count html
- implement js to enable background color change
- add boostrap styling to background html
- implement form and js to process form data and print to consoleX

stevensonmat2 committed:
Complete hw2:
- implement js to highlight text for count exercise
- add bootstrap styling to count html
- implement js to enable background color change
- add boostrap styling to background html
- implement form and js to process form data and print to consoleX

nicolekurtz93 commented on hw2/04-background.html:
```diff
@@ -19,10 +19,10 @@ <h1>Exercise 04 - Background</h1>
       </p>
       <div>
         <label>Background change interval (seconds): </label>
```
To make this more accessible, the label should be connected to the input field with a 'for'.

nicolekurtz93 commented on hw2/04-background.js:
```diff
@@ -1 +1,40 @@
 // Add your code here
```
The assignment sheet wants you to start the page with the background already changing every 3 seconds. You will need a page load event listener.

nicolekurtz93 commented on hw2/04-background.js:
```diff
@@ -1 +1,40 @@
+    clearInterval(intervalID);
+    buttonClasses.add('btn-primary');
+    buttonClasses.remove('btn-danger');
+    button.textContent = 'start';
```
For consistency, I would either change the start/stop text content to be capitalized or change the capitalization in your HTML file.

nicolekurtz93 commented on hw2/04-background.js:
```diff
@@ -1 +1,40 @@
+function handleButtonClick() {
+  const interval = input.value;
+  const buttonClasses = button.classList;
+
```
You might consider adding a check to make sure the user input is a number. I know you set the field to be type number, but in some browsers this can be ignored and users are able to add letters.

capstone-psu-2023 replied:
whaaaa kinda crap browsers do that lol

nicolekurtz93 commented on hw2/04-background.js:
```diff
@@ -1 +1,40 @@
+  const buttonClasses = button.classList;
+
+  if (!interval) {
+    return;
```
You might want to alert the user that they need to put in an input if we hit this if statement.

capstone-psu-2023 replied:
I like to keep them guessing

nicolekurtz93 commented on hw2/05-form.html:
```diff
@@ -10,10 +10,57 @@
+        </div>
+        <div class="row mb-3">
+          <label class="form-check-label" for="text-input">Anything else?</label>
+          <textarea type="text" name="comments" id="text-input" class="form-check-label"></textarea>
```
The spacing is a bit to close to the edge of the container for this textarea. You might want to add some spacing to the left.

nicolekurtz93 commented on hw2/05-form.js:
```diff
@@ -1 +1,24 @@
+    console.log(`${element.name}: ${element.value}`);
+  }
+
+  if (formElements.programmingLanguages) {
```
A possible rework would be to only output the name of the course if the course is selected. You could create a console.group for the courses and then include the name of all the courses selected in the group.

nicolekurtz93 commented on hw2/05-form.js:
```diff
@@ -1 +1,24 @@
+function handleSubmit(e) {
+  e.preventDefault();
+  const elementNames = ['full-name', 'email', 'registration-status', 'comments'];
+  const formElements = form.elements;
```
If you create a group that all of these console logs would be a part of, it would make it a bit more clean in the Console.

nicolekurtz93 commented on hw2/05-form.html:
```diff
@@ -10,10 +10,57 @@
+  <body class="bg-primary container mt-5">
+    <main class=" bg-light rounded p-4 d-flex flex-column max-sm-75">
       <form>
-        <h1>Exercise 05 - Form</h1>
```
Looks like you accidentally removed the form header. This is giving you an accessibility issue because there is no Header.

nicolekurtz93 reviewed (changes requested):
Great job! You had a couple accessibility issues, and were missing a component for the background exercise that I would suggest correcting.

xuvea commented on hw2/03-count.html:
```diff
@@ -9,16 +9,16 @@
-  <body>
-    <main>
-      <h1>Exercise 03 - Count</h1>
+  <body class="bg-primary container mt-5">
```
Instead of using the `bg-primary` for the background color I would use `style= "background-color: #01435C;"` for matching the color in the example pictures.

xuvea commented on hw2/03-count.html:
```diff
@@ -9,16 +9,16 @@
+      <h1 class="d-flex justify-content-center">Exercise 03 - Count</h1>
       <hr />
-      <div>
+      <div class="d-flex justify-content-center">
```
Content here shouldn't be centered, as it is aligned on the left in the example photo. Just have to remove `justify-content-center` to fix that

xuvea commented on hw2/03-count.js:
```diff
@@ -1,3 +1,18 @@
+  }
+  const highlightedText = searchableText.replaceAll(
+    searchValue,
+    `<mark class='font-weight-normal' style='background-color: yellow;'>${searchValue}</mark>`,
```
Instead of using mark for highlighting the matching text I would recommend using `<span style='background-color: yellow;></span>`, as it will not make the letters spaces weirdly apart. I also don't think the font weight is needed either as it doesn't appear to be doing anything.

xuvea commented on hw2/04-background.html:
```diff
@@ -9,8 +9,8 @@
   </head>
-  <body>
-    <main>
+  <body class="container mt-5">
+    <main class=" bg-light rounded p-4 d-flex flex-column max-sm-75">
```
Sizing of the container is a little strange here, as it doesn't match the example, try removing the classes from body and replace the classes in main with `card d-flex w-50 mx-auto mt-5 p-3`.

xuvea commented on hw2/04-background.html:
```diff
@@ -19,10 +19,10 @@ <h1>Exercise 04 - Background</h1>
       </div>
       <div>
-        <input />
+        <button type="button" id="start-stop" class="btn btn-primary">Start</button>
```
To better match the styling, you could add the class `mt-3` to the button.

xuvea commented on hw2/05-form.html:
```diff
@@ -10,10 +10,57 @@
       <form>
-        <h1>Exercise 05 - Form</h1>
+        <div class="mb-3">
+          <label for="fullName" class="form-label">Full name</label>
```
Label needs to be in bold, can use the class `fw-bold` to do this. Labels on line 21, 25, 34, and 55 also need to be bolded.

xuvea commented on hw2/05-form.html:
```diff
@@ -10,10 +10,57 @@
+          </select>
+        </div>
+        <div class="form-group">
+          <legend>Have you taken any of the following courses?</legend>
```
You could use a label here instead of a legend to properly label your form check.

stevensonmat2 committed:
update with changes based on feedback:
- address styling issues in html files to address accessiblity concerns
- make color change begin when page is loaded for ex. 4
- improve form styling
